gnucash/gnome/window-reconcile.c

```diff
@@ -1239,6 +1239,59 @@ gnc_reconcile_window_create_view_box(Account *account,
 }
 
 
+/** Get the split that should be selected after the current split is
+ *   deleted, so that the working position in the debit or credit view
+ *   is maintained. This will be the next split if it exists (so that
+ *   it is possible to delete multiple splits just be clicking the
+ *   Delete button multiple times), otherwise the previous split.
+ *   The returned split must be for a different transaction as the
+ *   Delete button will delete all splits in the transaction.
+ *
+ *  @param view The view to use.
+ *
+ *  @param split The split to be removed
+ */
+static Split *
+gnc_reconcile_window_get_next_split(GNCReconcileView *view, Split *split)
+{
+    gboolean valid;
+    gpointer pointer;
+    GtkTreeIter iter;
+    Split *next_split=NULL, *temp_split;
+    Transaction *trans=NULL;
+    GtkTreeModel* model = gtk_tree_view_get_model (GTK_TREE_VIEW (view));
+
+    trans = xaccSplitGetParent (split);
+    valid = gtk_tree_model_get_iter_first (model, &iter);
+    while (valid)
+    {
+        // Walk through the model list, reading each row
+        gtk_tree_model_get (model, &iter, REC_POINTER, &pointer, -1);
+        if(pointer == split)
+        {
+            // find next split from a different transaction
+            while (valid)
+            {
+                valid = gtk_tree_model_iter_next (model, &iter);
+                if (valid)
+                {
+                    gtk_tree_model_get (model, &iter, REC_POINTER, &temp_split, -1);
+                    if (trans != xaccSplitGetParent (temp_split))
+                        return temp_split;
+                }
+            }
+        }
+        if (valid)
+        {
+            if (trans != xaccSplitGetParent (pointer))
+                next_split = pointer;
+            valid = gtk_tree_model_iter_next (model, &iter);
+        }
+    }
+    return next_split;
```

Comment:
Fixed in PR 707 (against master). The deletion changes in this PR are more to do with maintaining visibility of where you're working than enabling faster multiple transaction deletes.
Without my changes, everytime the list refreshes, it would scroll to the top, even though a selection would still exist if the selected split had not been deleted.

Re the nested loops: I have fixed and must remember not to copy old code.